src/transformers/trainer.py

```diff
@@ -679,6 +679,11 @@ def __init__(
         # returned to 0 every time flos need to be logged
         self.current_flos = 0
         self.hp_search_backend = None
+        if self.model.__class__.name__ == "PeftModel" and self.args.label_names is None:
```

Comment:
I don't think this check is robust as the model could be e.g. `PeftForCausalLM`. Within this module we have `_is_peft_model` which I think we should use.

Reply:
ok, very good suggestion!